controllers/gatekeeper_controller.go

```diff
@@ -288,10 +293,45 @@ func (r *GatekeeperReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 
 // SetupWithManager sets up the controller with the Manager.
 func (r *GatekeeperReconciler) SetupWithManager(mgr ctrl.Manager, fromCPSMgrSource source.Source) error {
+	// Delete-only predicate for specific, cluster-scoped RBAC managed by this operator
+	deleteOnlyClusterRBAC := predicate.Funcs{
+		CreateFunc:  func(_ event.CreateEvent) bool { return false },
+		UpdateFunc:  func(_ event.UpdateEvent) bool { return false },
+		GenericFunc: func(_ event.GenericEvent) bool { return false },
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			name := e.Object.GetName()
+
+			return name == "gatekeeper-manager-role" || name == "gatekeeper-manager-rolebinding"
+		},
+	}
+
+	// Reusable delete-only, namespace-scoped predicate for namespaced resources
+	nsDeleteOnly := builder.WithPredicates(predicate.Funcs{
+		CreateFunc:  func(_ event.CreateEvent) bool { return false },
+		UpdateFunc:  func(_ event.UpdateEvent) bool { return false },
+		GenericFunc: func(_ event.GenericEvent) bool { return false },
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return e.Object.GetNamespace() == r.Namespace
+		},
+	})
+
 	return ctrl.NewControllerManagedBy(mgr).
 		WithOptions(controller.Options{MaxConcurrentReconciles: 1}).
 		Named("gatekeeper_reconciler").
 		For(&operatorv1alpha1.Gatekeeper{}).
+		// WebhookConfigurations are cluster-scoped and get updated frequently
+		// by CABundle injection controllers.
+		// If we Own() them, every injection update would enqueue a reconcile, creating a lot of churn.
+		Owns(&appsv1.Deployment{}, nsDeleteOnly).
+		Owns(&corev1.Service{}, nsDeleteOnly).
+		Owns(&corev1.Secret{}, nsDeleteOnly).
+		Owns(&corev1.ServiceAccount{}, nsDeleteOnly).
+		Owns(&policyv1.PodDisruptionBudget{}, nsDeleteOnly).
+		Owns(&rbacv1.Role{}, nsDeleteOnly).
+		Owns(&rbacv1.RoleBinding{}, nsDeleteOnly).
+		// Watch cluster-scoped RBAC delete events only (avoid update churn)
+		Owns(&rbacv1.ClusterRole{}, builder.WithPredicates(deleteOnlyClusterRBAC)).
+		Owns(&rbacv1.ClusterRoleBinding{}, builder.WithPredicates(deleteOnlyClusterRBAC)).
```

Comment:
`deleteOnlyClusterRBAC` is just the `predicate.Funcs`, but `nsDeleteOnly` is a bit nicer because it includes the `builder.WithPredicates` call. Is there a reason they are different? It seems like consistency would be nice so it could look like:

```suggestion
		Owns(&rbacv1.Role{}, nsDeleteOnly).
		Owns(&rbacv1.RoleBinding{}, nsDeleteOnly).
		// Watch cluster-scoped RBAC delete events only (avoid update churn)
		Owns(&rbacv1.ClusterRole{}, deleteOnlyClusterRBAC).
		Owns(&rbacv1.ClusterRoleBinding{}, deleteOnlyClusterRBAC).
```

Reply:
Thanks a lot

Reply:
what you think we only recreate the resource when it is deleted. Should I watch "UPDATE" to block user's update?

Reply:
For v3.20, I like this targeted solution that directly addresses the bug since it'll be going into a z-stream, and I think backporting it to v3.19 is reasonable as a result. For v3.21, we can add update operations.

Reply:
Great! Thanks @dhaiducek @JustinKuli

Reply:
I was concerned about performance and resource usage, so I only added support for DELETE.